compiler/rustc_target/src/spec/targets/armv7a_vex_v5_linker_script.ld

```diff
@@ -0,0 +1,58 @@
+OUTPUT_FORMAT("elf32-littlearm")
+
+ENTRY(_start)
+
+__user_ram_start = 0x03800000;
+__user_ram_length = 0x04800000;
+__user_ram_end = __user_ram_start + __user_ram_length;
+
+MEMORY {
+    USER_RAM : ORIGIN = __user_ram_start, LENGTH = __user_ram_length
+}
+
+__stack_length = 0x400000;
+/*
+It's currently unclear why subtracting anything is necessary, but it fixes memory permission errors.
+0x100 is an arbitrary number that works.
+*/
+__heap_end = __user_ram_end - __stack_length - 0x100;
```

Comment:
I think we should really either fix this or figure out why it happens before PRing upstream.

Reply:
Agreed.